apps/wallet/src/background/keyring/VaultStorage.ts

```diff
@@ -139,17 +140,18 @@ class VaultStorageClass {
     public async importKeypair(
         keypair: ExportedKeypair,
         password: string,
-        existingAccounts: Account[]
+        existingAccounts: (ImportedAccount | DerivedAccount)[]
     ) {
         if (!this.#vault) {
             throw new Error('Error, vault is locked. Unlock the vault first.');
         }
         const keypairToImport = fromExportedKeypair(keypair);
         const importedAddress = keypairToImport.getPublicKey().toSuiAddress();
-        const isDuplicate = existingAccounts.some(
-            (anAccount) =>
-                anAccount.publicKey.toSuiAddress() === importedAddress
-        );
+        const isDuplicate = existingAccounts.some((anAccount) => {
+            const suiAddress =
+                anAccount.accountKeypair.publicKey.toSuiAddress();
```

Comment:
Maybe we should make the address an accessory to minimize some of this logic.

Reply:
Good call!